include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -94,13 +95,22 @@ class bi_fm_index
 
 public:
 
+    static_assert(dimension_v<text_t> == 1 || dimension_v<text_t> == 2,
+                  "Only texts or collections of texts can be indexed.");
+
+    //!\brief Indicates whether index is built over a collection.
+    static bool constexpr is_collection = dimension_v<text_t> == 2;
+
     /*!\name Text types
      * \{
      */
     //!\brief The type of the forward indexed text.
     using text_type = text_t;
     //!\brief The type of the forward indexed text.
-    using rev_text_type = decltype(view::reverse(*text));
+    using rev_text_type = std::conditional_t<is_collection,
+                                            decltype(*text | view::deep{view::reverse} | view::deep{view::persist}
```

Comment:
Good catch 👍 

I know that I needed it there and that otherwise it wouldn't compile.
I probably changed something in the `fm_index` and did not revisit the `rev_text_type`.

Reply:
Actually I need it, because otherwise the `to_rev_iterator` function of the `bi_fm_index` cannot use the view.
I don't really understand why, because it's just a long stack trace, complaining that it cannot bind it without the `view::persist`.

Reply:
Ok, we need to investigate this. Can you please add a card for this on the project site for the search module